fastly/platform_tls.go

```diff
@@ -0,0 +1,350 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// GetPrivateKeyInput is an input to the GetPrivateKey function.
+// Allowed values for the fields are described at https://docs.fastly.com/api/platform-tls.
+type GetPrivateKeyInput struct {
+	ID string
+}
+
+// PrivateKey represents a private key is used to sign a Certificate.
+type PrivateKey struct {
```

Comment:
I'd be in favor of moving the `PrivateKey` structs and related requests into their own file. The APIs aren't specific to Platform TLS and are shared with the Fastly TLS product, as another reviewer noted.